### PR TITLE
feat(connector): [DEUTSCHEBANK] Integrate SEPA Payments

### PR DESCRIPTION
**Pull request by deepanshu-iiitu**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Integrate SEPA Payments for connector Deutsche Bank
https://testmerch.directpos.de/rest-api/apidoc/v2.1/index.html#chap_sdd

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

https://github.com/juspay/hyperswitch/issues/5858

## How did you test it?

1. Authorize:
Request:
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_2QpEeK0ByPRkb2DPBxHfad4i1WxQO3hZSuVzlFcg9mTrLFUHRpGfE0SCuQwZjGEb' \
--data '{
    "amount": 9000,
    "currency": "EUR",
    "confirm": true,
    "profile_id": "pro_EuXuxP2KVO0m1tlDWSau",
    "customer_id": "customer123",
    
    "payment_method": "bank_debit",
    "payment_method_type": "sepa",
    "payment_method_data": {
        "bank_debit": {
            "sepa_bank_debit": {
                "iban": "DE87123456781234567890"
            }
        }
    },
    "billing": {
        "address": {
            "first_name": "joseph",
            "last_name": "Doe"
        }
    },
}'
```

Response:
```
{
    "payment_id": "pay_9SIimBI1UXiGdsR4XCmH",
    "merchant_id": "merchant_1725539597",
    "status": "requires_customer_action",
    "amount": 9000,
    "net_amount": 9000,
    "amount_capturable": 9000,
    "amount_received": null,
    "connector": "deutschebank",
    "client_secret": "pay_9SIimBI1UXiGdsR4XCmH_secret_s0dXk9MNEsKru7clJoNK",
    "created": "2024-09-12T10:41:39.816Z",
    "currency": "EUR",
    "customer_id": "customer123",
    "customer": {
        "id": "customer123",
        "name": "John Doe",
        "email": "customer@gmail.com",
        "phone": "9999999999",
        "phone_country_code": "+1"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": null,
    "payment_method": "bank_debit",
    "payment_method_data": {
        "bank_debit": {
            "sepa": {
                "iban": "DE871************67890",
                "bank_account_holder_name": null
            }
        },
        "billing": null
    },
    "payment_token": "token_IORpLx2lNN9VK64COwy0",
    "shipping": null,
    "billing": {
        "address": {
            "city": null,
            "country": null,
            "line1": null,
            "line2": null,
            "line3": null,
            "zip": null,
            "state": null,
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "customer@gmail.com",
    "name": "John Doe",
    "phone": "9999999999",
    "return_url": "https://example.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": {
        "type": "redirect_to_url",
        "redirect_to_url": "http://localhost:8080/payments/redirect/pay_9SIimBI1UXiGdsR4XCmH/merchant_1725539597/pay_9SIimBI1UXiGdsR4XCmH_1"
    },
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "sepa",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "customer123",
        "created_at": 1726137699,
        "expires": 1726141299,
        "secret": "epk_a677e949924b417cb73e1badf1ac05f8"
    },
    "manual_retry_allowed": null,
    "connector_transaction_id": null,
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_EuXuxP2KVO0m1tlDWSau",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_MZAsrTowXuzpeXl4DMj2",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-09-12T10:56:39.816Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-09-12T10:41:42.563Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```

Then do redirection

2. PSYNC:
Request:
```
curl --location 'http://localhost:8080/payments/pay_aCYBEjtBmw7l7u4jQWhM' \
--header 'Accept: application/json' \
--header 'api-key: dev_2QpEeK0ByPRkb2DPBxHfad4i1WxQO3hZSuVzlFcg9mTrLFUHRpGfE0SCuQwZjGEb'
```

Response:
```
{
    "payment_id": "pay_aCYBEjtBmw7l7u4jQWhM",
    "merchant_id": "merchant_1725539597",
    "status": "succeeded",
    "amount": 9000,
    "net_amount": 9000,
    "amount_capturable": 0,
    "amount_received": 9000,
    "connector": "deutschebank",
    "client_secret": "pay_aCYBEjtBmw7l7u4jQWhM_secret_4TLRv0W2zGPEaN5hxwOC",
    "created": "2024-09-12T10:32:33.053Z",
    "currency": "EUR",
    "customer_id": "customer123",
    "customer": {
        "id": "customer123",
        "name": "John Doe",
        "email": "customer@gmail.com",
        "phone": "9999999999",
        "phone_country_code": "+1"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": null,
    "payment_method": "bank_debit",
    "payment_method_data": {
        "bank_debit": {
            "sepa": {
                "iban": "DE871************67890",
                "bank_account_holder_name": null
            }
        },
        "billing": null
    },
    "payment_token": "token_1gP6ixqxxYORXgW6mAxq",
    "shipping": null,
    "billing": {
        "address": {
            "city": null,
            "country": null,
            "line1": null,
            "line2": null,
            "line3": null,
            "zip": null,
            "state": null,
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "customer@gmail.com",
    "name": "John Doe",
    "phone": "9999999999",
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "sepa",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": null,
    "manual_retry_allowed": false,
    "connector_transaction_id": "fNo3C3goIky4BNzSU7ys9n",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_EuXuxP2KVO0m1tlDWSau",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_MZAsrTowXuzpeXl4DMj2",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-09-12T10:47:33.053Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-09-12T10:40:37.356Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```

3. Capture: 
Request:
```
curl --location 'http://localhost:8080/payments/pay_zwqrtx6EMgaV2gkK7WdS/capture' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_2QpEeK0ByPRkb2DPBxHfad4i1WxQO3hZSuVzlFcg9mTrLFUHRpGfE0SCuQwZjGEb' \
--data '{}'
```

Response:
```
{
    "payment_id": "pay_zwqrtx6EMgaV2gkK7WdS",
    "merchant_id": "merchant_1725539597",
    "status": "succeeded",
    "amount": 9000,
    "net_amount": 9000,
    "amount_capturable": 0,
    "amount_received": 9000,
    "connector": "deutschebank",
    "client_secret": "pay_zwqrtx6EMgaV2gkK7WdS_secret_Br19bUUvGCMLAiG1Fusv",
    "created": "2024-09-12T10:42:41.662Z",
    "currency": "EUR",
    "customer_id": "customer123",
    "customer": {
        "id": "customer123",
        "name": "John Doe",
        "email": "customer@gmail.com",
        "phone": "9999999999",
        "phone_country_code": "+1"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "manual",
    "payment_method": "bank_debit",
    "payment_method_data": {
        "bank_debit": {
            "sepa": {
                "iban": "DE871************67890",
                "bank_account_holder_name": null
            }
        },
        "billing": null
    },
    "payment_token": "token_RzJMp7J8sp2wWoRnBYPA",
    "shipping": null,
    "billing": {
        "address": {
            "city": null,
            "country": null,
            "line1": null,
            "line2": null,
            "line3": null,
            "zip": null,
            "state": null,
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "customer@gmail.com",
    "name": "John Doe",
    "phone": "9999999999",
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "sepa",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": null,
    "manual_retry_allowed": false,
    "connector_transaction_id": "8v739jbpsaKxnsCB52QmeF",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_EuXuxP2KVO0m1tlDWSau",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_MZAsrTowXuzpeXl4DMj2",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-09-12T10:57:41.662Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-09-12T10:42:57.446Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```

4. Void:
Request:
```
curl --location 'http://localhost:8080/payments/pay_sH3vFdwGOoPjobg4sud9/cancel' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_2QpEeK0ByPRkb2DPBxHfad4i1WxQO3hZSuVzlFcg9mTrLFUHRpGfE0SCuQwZjGEb' \
--data '{}'
```

Response:
```
{
    "payment_id": "pay_sH3vFdwGOoPjobg4sud9",
    "merchant_id": "merchant_1725539597",
    "status": "cancelled",
    "amount": 9000,
    "net_amount": 9000,
    "amount_capturable": 0,
    "amount_received": null,
    "connector": "deutschebank",
    "client_secret": "pay_sH3vFdwGOoPjobg4sud9_secret_jlUGjGn6ADDu02Ql8O4K",
    "created": "2024-09-12T10:44:13.912Z",
    "currency": "EUR",
    "customer_id": "customer123",
    "customer": {
        "id": "customer123",
        "name": "John Doe",
        "email": "customer@gmail.com",
        "phone": "9999999999",
        "phone_country_code": "+1"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "manual",
    "payment_method": "bank_debit",
    "payment_method_data": {
        "bank_debit": {
            "sepa": {
                "iban": "DE871************67890",
                "bank_account_holder_name": null
            }
        },
        "billing": null
    },
    "payment_token": "token_2Mq0WwprhovjSGXcJmZa",
    "shipping": null,
    "billing": {
        "address": {
            "city": null,
            "country": null,
            "line1": null,
            "line2": null,
            "line3": null,
            "zip": null,
            "state": null,
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "customer@gmail.com",
    "name": "John Doe",
    "phone": "9999999999",
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "sepa",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": null,
    "manual_retry_allowed": false,
    "connector_transaction_id": "9VFMneA8NhYUj7xoG0bmMA",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_EuXuxP2KVO0m1tlDWSau",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_MZAsrTowXuzpeXl4DMj2",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-09-12T10:59:13.912Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-09-12T10:44:39.083Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```

Note: Refunds are not working right now due to some issue with the connector account.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Narayanbhat166 — DISMISSED** [review]
Core changes Looks Good to me

**Review by SanchithHegde — DISMISSED** [review]

File: crates/hyperswitch_connectors/src/connectors/deutschebank.rs [hunk 1]
```diff
@@ -263,10 +492,18 @@ impl ConnectorIntegration<PSync, PaymentsSyncData, PaymentsResponseData> for Deu
...
-        _connectors: &Connectors,
+        req: &PaymentsSyncRouterData,
+        connectors: &Connectors,
     ) -> CustomResult<String, errors::ConnectorError> {
-        Err(errors::ConnectorError::NotImplemented("get_url method".to_string()).into())
+        let tx_id = req
+            .request
+            .connector_transaction_id
+            .get_connector_transaction_id()
+            .change_context(errors::ConnectorError::MissingConnectorTransactionID)?;
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank.rs:502]: Can we do this sync with our payment_id?

File: crates/hyperswitch_connectors/src/connectors/deutschebank.rs [hunk 2]
```diff
@@ -389,7 +640,86 @@ impl ConnectorIntegration<Capture, PaymentsCaptureData, PaymentsResponseData> fo
...
+        &self,
+        data: &PaymentsCancelRouterData,
+        event_builder: Option<&mut ConnectorEvent>,
+        res: Response,
+    ) -> CustomResult<PaymentsCancelRouterData, errors::ConnectorError> {
+        let response: deutschebank::DeutschebankPaymentsResponse = res
+            .response
+            .parse_struct("Deutschebank PaymentsCaptureResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        event_builder.map(|i| i.set_response_body(&response));
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank.rs:706]: Rename the parse_struct to `PaymentsCancelResponse`

File: crates/hyperswitch_connectors/src/connectors/deutschebank.rs (hunk 1 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank.rs:502]: No. Only with deutesche tx_id

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 3]
```diff
@@ -56,81 +128,317 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
...
-                    card,
-                })
-            }
+            PaymentMethodData::BankDebit(BankDebitData::SepaBankDebit { iban, .. }) => Ok(Self {
+                approval_by: DeutschebankSEPAApproval::Click,
+                email_address: item.router_data.request.get_email()?,
+                iban,
+                first_name: billing_address.get_first_name()?.clone(),
+                last_name: billing_address.get_last_name()?.clone(),
+            }),
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:139]: Add these in `defualt.rs` for dynamic fields

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 4]
```diff
@@ -56,81 +128,317 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
...
+            },
+            response: Ok(PaymentsResponseData::TransactionResponse {
+                resource_id: ResponseId::NoResponseId,
+                redirection_data: Some(RedirectForm::Form {
+                    endpoint: item.data.request.get_complete_authorize_url()?,
+                    method: common_utils::request::Method::Get,
+                    form_fields: HashMap::from([
+                        (
+                            "reference".to_string(),
+                            item.response.reference.unwrap_or("".to_string()),
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:231]: If reference is a critical fields as if its not available you fail the payment in complete_authorize, its better to fail it here itself

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 5]
```diff
@@ -56,81 +128,317 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
...
+            CompleteAuthorize,
+            DeutschebankPaymentsResponse,
+            CompleteAuthorizeData,
+            PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
+        let resource_id = ResponseId::ConnectorTransactionId(
+            item.response
+                .tx_id
+                .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?,
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:429]: If connector_transaction_id is missing shall we fail the payment or mark it pending?

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 6]
```diff
@@ -144,87 +452,233 @@ impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, Paymen
...
+impl TryFrom<&DeutschebankRouterData<&PaymentsCaptureRouterData>> for DeutschebankCaptureRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &DeutschebankRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &DeutschebankRouterData<&PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
         Ok(Self {
-            amount: item.amount.to_owned(),
+            changed_amount: item.amount,
+            kind: "DIRECTDEBIT".to_string(),
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:468]: Create an enum for this

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 7]
```diff
@@ -144,87 +452,233 @@ impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, Paymen
...
+            Capture,
+            DeutschebankPaymentsResponse,
+            PaymentsCaptureData,
+            PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let resource_id = ResponseId::ConnectorTransactionId(
+            item.response
+                .tx_id
+                .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?,
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:495]: same here

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 8]
```diff
@@ -144,87 +452,233 @@ impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, Paymen
...
+#[derive(Default, Debug, Serialize)]
+pub struct DeutschebankReversalRequest {
+    kind: String,
+}
+
+impl TryFrom<&PaymentsCancelRouterData> for DeutschebankReversalRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            kind: "DIRECTDEBIT".to_string(),
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:577]: Move it the same enum as above

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 9]
```diff
@@ -144,87 +452,233 @@ impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, Paymen
...
+    changed_amount: MinorUnit,
+    kind: String,
+}
+
+impl<F> TryFrom<&DeutschebankRouterData<&RefundsRouterData<F>>> for DeutschebankRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &DeutschebankRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            changed_amount: item.amount.to_owned(),
+            kind: "DIRECTDEBIT".to_string(),
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:611]: same here

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs [hunk 10]
```diff
@@ -144,87 +452,233 @@ impl<F, T> TryFrom<ResponseRouterData<F, DeutschebankPaymentsResponse, T, Paymen
...
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: RefundsResponseRouterData<RSync, RefundResponse>,
+        item: RefundsResponseRouterData<Execute, DeutschebankPaymentsResponse>,
     ) -> Result<Self, Self::Error> {
+        let connector_refund_id = item
+            .response
+            .tx_id
+            .ok_or(errors::ConnectorError::MissingConnectorRefundID)?;
```
Comment (SamraatBansal) [inline crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:626]: same here

**Review by SamraatBansal — COMMENTED** [review]
Add the configurations in connector_configs for the dashboard.

Also please update the sheet for tracking connector feature set

File: crates/hyperswitch_connectors/src/connectors/deutschebank.rs (hunk 2 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank.rs:706]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 4 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:231]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 5 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:429]: We are not marking tx_id as option now. Deserialization will happen if tx_id is not in response.

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 6 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:468]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 7 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:495]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 8 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:577]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 9 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:611]: Done

File: crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs (hunk 10 above)
Comment (deepanshu-iiitu) [reply crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs:626]: Done

File: crates/router/src/configs/defaults.rs [hunk 11]
```diff
@@ -11209,8 +11209,43 @@ impl Default for super::settings::RequiredFields {
...
+                                            (
+                                                "billing.address.last_name".to_string(),
+                                                RequiredFieldInfo {
+                                                    required_field: "payment_method_data.billing.address.last_name".to_string(),
+                                                    display_name: "owner_name".to_string(),
+                                                    field_type: enums::FieldType::UserBillingName,
+                                                    value: None,
+                                                }
+                                            ),
+                                            (
```
Comment (SamraatBansal) [inline crates/router/src/configs/defaults.rs:11236]: These fields are for mandate payments ,add it in common not in non_mandate

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**